earthfile2llb/converter.go

```diff
@@ -2535,29 +2659,86 @@ func (c *Converter) processNonConstantBuildArgFunc(ctx context.Context) variable
 	}
 }
 
-func (c *Converter) vertexPrefix(ctx context.Context, cmdID int, local bool, interactive bool, internal bool, secrets []string) string {
+func (c *Converter) newLogbusCommand(ctx context.Context, name string) (string, *logbus.Command, error) {
+
+	cmdID := fmt.Sprintf("%s/%d", c.mts.Final.ID, c.newCmdID())
+
+	var gitURL, gitHash, fileRelToRepo string
+	if c.gitMeta != nil {
+		gitURL = c.gitMeta.RemoteURL
+		gitHash = c.gitMeta.Hash
+		fileRelToRepo = path.Join(c.gitMeta.RelDir, "Earthfile")
+	}
+
+	platform := c.platr.Materialize(c.platr.Current())
+
+	cmd, err := c.opt.Logbus.Run().NewCommand(
+		cmdID,
+		name,
+		c.mts.Final.ID,
+		c.mts.Final.Target.String(),
+		platform.String(),
+		false, // cached
+		false, // local
+		false, // interactive
+		SourceLocationFromContext(ctx),
+		gitURL,
+		gitHash,
+		fileRelToRepo,
+	)
+	if err != nil {
+		return "", nil, err
+	}
+
+	cmd.SetStart(time.Now())
+
+	return cmdID, cmd, nil
+}
+
+func (c *Converter) newLogbusCommandFromVertex(ctx context.Context, vm *vertexmeta.VertexMeta) error {
+	_, err := c.opt.Logbus.Run().NewCommand(
+		vm.CommandID,
+		"unknown",
+		c.mts.Final.ID,
+		c.mts.Final.Target.String(),
+		vm.Platform,
+		false, // cached
+		vm.Local,
+		vm.Interactive,
+		SourceLocationFromContext(ctx),
+		vm.RepoGitURL,
+		vm.RepoGitHash,
+		vm.RepoFileRelToRepo,
+	)
+	return err
+}
+
+func (c *Converter) vertexMeta(ctx context.Context, cmdID int, local, interactive, internal bool, secrets []string) *vertexmeta.VertexMeta {
```

Comment:
What if we change this to `newVertexMeta` and instead generate the new `cmdID` within? It would simplify the API / call sequence of this a little.

Reply:
I started with that but decided to separate the two for reasons I can't quite recall... :sweat_smile: 

I'll take a look at combining them. Perhaps it was just that they seem like disparate operations.

Reply:
All good if you find that they need to be called independently too - I didn't look carefully.

Reply:
I think that was my original thinking, but as it happens they are always called together. I'll merge them.

Reply:
Updated.